src/taxonomy/TaxonomyListPage.jsx

```diff
@@ -80,18 +85,78 @@ const TaxonomyListPage = () => {
     });
   }, [setToastMessage]);
 
+  const {
+    data: organizationListData,
+    isSuccess: isOrganizationListLoaded,
+  } = useOrganizationListData();
+
   const useTaxonomyListData = () => {
-    const taxonomyListData = useTaxonomyListDataResponse();
-    const isLoaded = useIsTaxonomyListDataLoaded();
+    const taxonomyListData = useTaxonomyListDataResponse(selectedOrgFilter);
+    const isLoaded = useIsTaxonomyListDataLoaded(selectedOrgFilter);
     return { taxonomyListData, isLoaded };
   };
   const { taxonomyListData, isLoaded } = useTaxonomyListData();
 
-  const getOrgSelect = () => (
+  const isOrgSelected = (value) => (value === selectedOrgFilter ? <Check /> : null);
+
+  const getOrgSelect = () => {
     // Organization select component
-    // TODO Add functionality to this component
-    undefined
-  );
+
+    const selectOptions = [
+      <MenuItem
+        key="all-orgs-taxonomies"
+        className="x-small"
+        iconAfter={() => isOrgSelected('All taxonomies')}
+        onClick={() => setSelectedOrgFilter('All taxonomies')}
+      >
+        { isOrgSelected('All taxonomies')
+          ? intl.formatMessage(messages.orgInputSelectDefaultValue)
+          : intl.formatMessage(messages.orgAllValue)}
+      </MenuItem>,
+      <MenuItem
+        key="unassigned-taxonomies"
+        className="x-small"
+        iconAfter={() => isOrgSelected('Unassigned')}
+        onClick={() => setSelectedOrgFilter('Unassigned')}
+      >
+        { intl.formatMessage(messages.orgUnassignedValue) }
+      </MenuItem>,
+    ];
+
+    if (isOrganizationListLoaded && organizationListData) {
+      organizationListData.map(org => (
+        selectOptions.push(
+          <MenuItem
+            key={`${org}-taxonomies`}
+            className="x-small"
+            iconAfter={() => isOrgSelected(org)}
+            onClick={() => setSelectedOrgFilter(org)}
+          >
+            {org}
+          </MenuItem>,
+        )
+      ));
+    }
+
+    return (
+      <SelectMenu
```

Comment:
Agreed, updated here: https://github.com/openedx/frontend-app-course-authoring/pull/755/commits/c42e0892121655f4fa54a950fe2bef99614e5716 However I kept the `getOrgSelect` to initialize the component, since I needed to pass in some props